src/lib/operators/table_scan.cpp

```diff
@@ -0,0 +1,247 @@
+      return [](T left, T right) -> bool { return left > right; };
+    case ScanType::OpGreaterThanEquals:
+      return [](T left, T right) -> bool { return left >= right; };
+    default:
+      throw std::runtime_error("ScanType is not defined");
+  }
+}
+
+template <>
+std::function<bool(std::string, std::string)> get_comparator(ScanType type) {  // std::function<bool(T, T)>
+  switch (type) {
+    case ScanType::OpEquals:
+      return [](std::string left, std::string right) -> bool { return (left.compare(right) == 0); };
+    case ScanType::OpNotEquals:
+      return [](std::string left, std::string right) -> bool { return (left.compare(right) != 0); };
+    case ScanType::OpLessThan:
+      return [](std::string left, std::string right) -> bool { return (left.compare(right) < 0); };
+    case ScanType::OpLessThanEquals:
+      return [](std::string left, std::string right) -> bool { return (left.compare(right) <= 0); };
+    case ScanType::OpGreaterThan:
+      return [](std::string left, std::string right) -> bool { return (left.compare(right) > 0); };
+    case ScanType::OpGreaterThanEquals:
+      return [](std::string left, std::string right) -> bool { return (left.compare(right) >= 0); };
+    default:
+      throw std::runtime_error("ScanType is not defined");
+  }
+}
+}  // namespace
+
+namespace opossum {
+
+TableScan::TableScan(const std::shared_ptr<const AbstractOperator> in, ColumnID column_id, const ScanType scan_type,
+                     const AllTypeVariant search_value)
+    : _in(in), _column_id(column_id), _scan_type(scan_type), _search_value(search_value) {}
+
+std::shared_ptr<const Table> TableScan::_on_execute() {
+  // Get input table
+  const std::shared_ptr<const Table> input_table = _in->get_output();
+
+  // get chunk count
+  const ChunkID input_chunk_count = input_table->chunk_count();
+
+  // create PosList
+  const auto pos_list = std::make_shared<PosList>();
+
+  // cast all segments to ValueSegment or DictionarySegment
+  const auto type_name = input_table->column_type(_column_id);
+
+  resolve_data_type(type_name, [&](auto type) {
+    // Deduce type name of column type
+    using Type = typename decltype(type)::type;
+    const auto comparator = get_comparator<Type>(_scan_type);
+    const auto typed_search_value = type_cast<Type>(_search_value);
+
+    // for each chunk and the relevant segment, compare the values and append to PosList
+    for (ChunkID chunk_index = ChunkID(0); chunk_index < input_chunk_count; ++chunk_index) {
+      const auto& segment = input_table->get_chunk(chunk_index).get_segment(_column_id);
+
+      if (const auto value_segment = std::dynamic_pointer_cast<ValueSegment<Type>>(segment)) {
+        // if segment is value segment
+        // create a PosList-equivalent vector with every possible chunk_offset
+        std::vector<ChunkOffset> input_filter(value_segment->size());
+        // fill the input_filter vector with sequentially increasing values
```

Comment:
I see your point. Nice hint. I would fix this but because it would take some time that is better invested in our group project, we would not change this here.

Reply:
You are right, you should not invest any more time here. The comments are only feedback for you. There is absolutely no need to integrate any proposed changes.